src/db_impl.cc

```diff
@@ -818,6 +818,24 @@ void TitanDBImpl::ReleaseSnapshot(const Snapshot* snapshot) {
   // We can record here whether the oldest snapshot is released.
   // If not, we can just skip the next round of purging obsolete files.
   db_->ReleaseSnapshot(snapshot);
+  {
+    MutexLock l(&mutex_);
```

Comment:
It is not wise to acuqire a mutex here, since this is a hot path, almost all read requrests go through this (Titan creates a `ManagedSnapshot` implicitly). I will refactor this with atomic operations, but in a separate PR